ui/desktop/src/components/SessionHistoryView.tsx

```diff
@@ -0,0 +1,195 @@
+import MarkdownContent from './MarkdownContent';
+import ToolCallWithResponse from './ToolCallWithResponse';
+import { ToolRequestMessageContent, ToolResponseMessageContent } from '../types/message';
+
+interface SessionHistoryViewProps {
+  session: SessionDetails;
+  isLoading: boolean;
+  error: string | null;
+  onBack: () => void;
+  onResume: () => void;
+  onRetry: () => void;
+}
+
+const SessionHistoryView: React.FC<SessionHistoryViewProps> = ({
+  session,
+  isLoading,
+  error,
+  onBack,
+  onResume,
+  onRetry,
+}) => {
+  // Move the tool response mapping logic outside of the render loop
+  const getToolResponsesMap = (messageIndex: number, toolRequests: ToolRequestMessageContent[]) => {
+    const responseMap = new Map();
+
+    // Look for tool responses in subsequent messages
+    if (messageIndex >= 0) {
+      for (let i = messageIndex + 1; i < session.messages.length; i++) {
+        const responses = session.messages[i].content
+          .filter((c) => c.type === 'toolResponse')
+          .map((c) => c as ToolResponseMessageContent);
+
+        for (const response of responses) {
+          // Check if this response matches any of our tool requests
+          const matchingRequest = toolRequests.find((req) => req.id === response.id);
+          if (matchingRequest) {
+            responseMap.set(response.id, response);
+          }
+        }
+      }
+    }
+
+    return responseMap;
+  };
+
+  return (
+    <div className="h-screen w-full">
+      <div className="relative flex items-center h-[36px] w-full bg-bgSubtle"></div>
+
+      {/* Top Row - back, info, reopen thread (fixed) */}
+      <Card className="px-8 pt-6 pb-4 bg-bgSecondary flex items-center">
+        <BackButton showText={false} onClick={onBack} className="text-textStandard" />
+
+        {/* Session info row */}
+        <div className="ml-8">
+          <h1 className="text-lg font-bold text-textStandard">
+            {session.metadata.description || session.session_id}
+          </h1>
+          <div className="flex items-center text-sm text-textSubtle mt-2 space-x-4">
+            <span className="flex items-center">
+              <Clock className="w-4 h-4 mr-1" />
+              {new Date(session.messages[0]?.created * 1000).toLocaleString()}
+            </span>
+            <span className="flex items-center">
+              <MessageSquare className="w-4 h-4 mr-1" />
+              {session.metadata.message_count} messages
+            </span>
+            {session.metadata.total_tokens !== null && (
+              <span className="flex items-center">
+                {session.metadata.total_tokens.toLocaleString()} tokens
+              </span>
+            )}
+          </div>
+        </div>
+
+        <span
+          onClick={onResume}
+          className="ml-auto text-md cursor-pointer text-textStandard hover:font-bold hover:scale-105 transition-all duration-150"
+        >
+          Resume session
```

Comment:
nit: for other CTAs here we use title case, such as "Try Again" below. Should align.